### DISCUSSION TITLE
OTA Header - location of builddate in OTA-file?

### DISCUSSION THREAD

**Discussion by PatrickNL68** [Koenkk/zigbee2mqtt#12465] [Devices]
In the common.js-file (https://github.com/Koenkk/zigbee-herdsman-converters/tree/master/lib/ota) I can determine how the information is stored in the OTA-file. But I can not find the bytes where the builddate is stored... Where can I find this?

function parseImage(buffer) {
    const header = {
        otaUpgradeFileIdentifier: buffer.subarray(0, 4),
        otaHeaderVersion: buffer.readUInt16LE(4),
        otaHeaderLength: buffer.readUInt16LE(6),
        otaHeaderFieldControl: buffer.readUInt16LE(8),
        manufacturerCode: buffer.readUInt16LE(10),
        imageType: buffer.readUInt16LE(12),
        fileVersion: buffer.readUInt32LE(14),
        zigbeeStackVersion: buffer.readUInt16LE(18),
        otaHeaderString: buffer.toString('utf8', 20, 52),
        totalImageSize: buffer.readUInt32LE(52),
    };

Regards, Patrick

**Comment by Koenkk** [comment]
This info is not stored in the OTA header but in the firmware code itself (which differs per stack).
